### DISCUSSION TITLE
資料型態

### DISCUSSION THREAD

**Discussion by jerry850313** [1101-GS4719-C/Python-Community-GS4719-C#22] [General]
python相對C++而言，在資料型態上沒有所謂上限，
所以不像C++，在運算上可能會overflow，
而這點相對C++方便許多。
(以下連結分別連結到C++跟python的資料型態介紹)
[python](https://ithelp.ithome.com.tw/articles/10213724)
[C++](https://blog.xuite.net/tsai.oktomy/program/64004343)

**Comment by chia-kai-chang** [comment]
@jerry850313 發現的很棒
應該是在python裡面，**整數**資料型態的數值沒有上限

**Reply by chia-kai-chang** [reply]
Python中，**整數**資料型態所描述的數值沒有上限，取決於PC的記憶體

**Comment by EricAgumonJou** [comment]
剛剛試了一下發現挺有趣的是:

打>>2.0**(1023)
會顯示出:8.98846567431158e+307**

打>>2.0**(1024), 則會出現overflow
看來我的PC的記憶體不算大  : ) )
如下:

OverflowError                             Traceback (most recent call last)
~\AppData\Local\Temp/ipykernel_13968/2415494082.py in <module>
----> 1 2.0**1024 #溢位overflow

OverflowError: (34, 'Result too large')

**Reply by jerry850313** [reply]
再插幾張64GB就行了吧?

**Reply by chia-kai-chang** [reply]
因為是浮點數，所以有最大值限制喔
整數在Python則沒有最大值限制
